core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -163,10 +169,17 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
   val fetchMessageConversionsRate = newMeter(BrokerTopicStats.FetchMessageConversionsPerSec, "requests", TimeUnit.SECONDS, tags)
   val produceMessageConversionsRate = newMeter(BrokerTopicStats.ProduceMessageConversionsPerSec, "requests", TimeUnit.SECONDS, tags)
 
+  def removeMetricHelper(metricType: String, tags: scala.collection.Map[String, String]): Unit = {
+    if (metricTypeMap.contains(metricType)) {
```

Comment:
There's still a potential race condition between line 173 and 174: a thread can kicks in between and remove the metric.

I think we can just do:

```
val metric: Metric = metricTypeMap.remove(metricType)
if (metric != null) {
   removeMetric(metricTypeMap, ...);
}
```

Reply:
Noted